src/experimental/riff-raff-yaml-file/deployments/cloudformation.ts

```diff
@@ -62,8 +62,12 @@ export function addAmiParametersToCloudFormationDeployment(
       ...acc,
       [amiParameter.node.id]: {
         BuiltBy: "amigo",
-        Recipe: imageRecipe,
-        AmigoStage: "PROD",
+        AmigoStage: "PROD", //could be overridden below
+        ...(typeof imageRecipe === "string"
+          ? {
+              Recipe: imageRecipe,
+            }
+          : imageRecipe),
```

Comment:
I think this could be refactored to improve readability, and remove reliance on last write wins:

```ts
const { imageRecipe } = asg;
const amigoStage = typeof imageRecipe === "string" ? "PROD" : imageRecipe.AmigoStage;
const imageRecipeName = typeof imageRecipe === "string" ? "PROD" : imageRecipe.Recipe;

return {
  ...acc,
  [amiParameter.node.id]: {
    BuiltBy: "amigo",
    AmigoStage: amigoStage,
    Recipe: imageRecipeName
  }
}
```

As noted above, having `imageRecipe` be a `AmigoProps` only, would make this more readable still, at the cost of a breaking change.

Reply:
nice suggestion, that is more readable - I've just implemented it like that

Reply:
👍🏽  tests are failing due to ordering change 😅. Making it alphabetical would make future changes easier.